docs/misc/components.md

```diff
@@ -0,0 +1,105 @@
+        .withColor(0x0000FF)
+        .withItalic(true);
+Style red = Style.EMPTY // Copies empty style and sets color to red
+        .withColor(0xFF0000);
+Style bold = Style.EMPTY // Copies empty style and makes it bold
+        .withBold(true);
+
+Style doubleLines = Style.EMPTY // Copies empty style and makes it both underlined and strikethrough
+        .withUnderlined(true)
+        .withStrikethrough(true);
+
+text.setStyle(blue); // Sets style of the text to be blue and italic
+text.withStyle(red).withStyle(bold).withStyle(doubleLines); // Overwrites blue and italic style to be red, bold, underlined, and strikethrough
+```
+This results in red, bold text with two lines.
+
+Chaining Components
+===================
+
+`MutableComponent` can have additional components chained as siblings with `MutableComponent#append`. Chained components can be retrieved with `MutableComponent#getSiblings`.
+
+`Component` stores its siblings like a tree, and its style is inherited to its siblings.
+Below example will progressively apply style on top of each other:
+```java
+Style bold = Style.EMPTY.withBold(true);
+Style italic = Style.EMPTY.withItalic(true);
+Style strike = Style.EMPTY.withStrikethrough(true);
+Style underlined = Style.EMPTY.withUnderlined(true);
+
+MutableComponent a = Component.literal("a");
+MutableComponent b = Component.literal("b").setStyle(bold);
+MutableComponent c = Component.literal("c").setStyle(italic);
+MutableComponent d = Component.literal("d").setStyle(strike);
+MutableComponent e = Component.literal("e").setStyle(underlined);
+a.append(b);
+b.append(c);
+c.append(d);
+d.append(e);
+```
+[Here's how it looks like in-game][style_annotated]
+
+`ComponentContents`
+===================
+
+Component's text contents are handled by `ComponentContents`, it holds data and defines how to represent it as text.
+Here's the list of predefined `ComponentContents`:
+* `LiteralContents(String text)` - represents passed string literal as is.
+* `TranslatableContents(String translationKey, Object[] args)` - represents passed `translationkey` as localized text. `args` is inserted in place of format specifiers such as `%s` and `%1$s`, read [formatting] for more details.
+* `KeybindContents(String name)` - represents keybind name such as `tutorial.move.title` as name of Keyboard Key.
+* `NbtContents(String path, boolean interpreting, Optional<Component> separator, DataSource dataSource)` - represents data from `dataSource` specified by `path`. `separator` will be used when there are multiple entities.
+* `ScoreContents(String name, String objective)` - represents `objectives`'s score of entities selected by [entity selector][selectors] `name`  .
+* `SelectorContents(String pattern, Optional<Component> separator)` - represents entity selector specified by [entity selector][selectors] `pattern` as a list of selected entities' display names. `separator` will be used when there are multiple entities.
```

Comment:
You already mentioned the static helpers above, this section seems redundant and probably should be mentioned above instead.